src/bin/pg_autoctl/file_utils.c

```diff
@@ -742,6 +742,21 @@ set_program_absolute_path(char *program, int size)
 }
 
 
+/*
+ * normalize_filename_inplace returns the real path of a given filename that
+ * belongs to an existing file on-disk, resolving symlinks and pruning
+ * double-slashes and other weird constructs. It stores it back in filename,
+ * which is assumed to be a buffer of size MAXPGPATH.
+ */
+bool
+normalize_filename_inplace(char *filename)
```

Comment:
Can we instead make it so that `normalise_filename` works well when given the same filename twice, same as `join_path_components` that we use that way (in e.g. src/bin/pg_autoctl/config.c):

``` c
	/* and finally add the configuration file name */
	join_path_components(filename, filename, name);

	/* normalize the path to the configuration file, if it exists */
	if (!normalize_filename(filename, dst, MAXPGPATH))
	{
		/* errors have already been logged */
		return false;
	}
```

Reply:
Modified, it actually already supported that.